test/splitOnRegex.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+
+
+test('splitOnRegex', function() {
+
+  eq(typeof S.splitOnRegex, 'function');
+  eq(S.splitOnRegex.length, 2);
+  eq(S.splitOnRegex.toString(), 'splitOnRegex :: RegExp -> String -> Array String');
+
+  eq(S.splitOnRegex(/b/, 'abc'), ['a', 'c']);
+  eq(S.splitOnRegex(/d/, 'abc'), ['abc']);
+  eq(S.splitOnRegex(/b(?=c)/, 'abcde'), ['a', 'cde']);
```

Comment:
This doesn't seem much different to the `/b/` regex. Maybe apply it to something like `abc abd`?

```js
 eq(S.splitOnRegex(/b(?=c)/, 'abc abd'), ['a', 'c abd']);
```